cli/JSXV4.md

```diff
@@ -76,15 +76,95 @@ module Wrapper = {
 let make = () => body
 ```
 
-**V4 Spec**
+### Migration of V3 components that depend on the internal representation
+
+Some components in existing projects are written in a way that is dependent on the V3 internal representation.
+Here are a few examples of how to convert them to V4.
+
+#### `makeProps` does not exist in V4
+
+Rewrite this:
+
+```rescript
+// V3
+module M = {
+  @obj external makeProps: (~msg: 'msg, ~key: string=?, unit) => {"msg": 'msg} = "" // No more makeProps
+
+  let make = (~msg) => {
+    <div> {React.string(msg)} </div>
+  }
+}
+```
+
+To this:
+
+```rescript
+// V4
+module M = {
+  type props<'msg> = {msg: 'msg}
+  let make = props => <div> {React.string(props.msg)} </div>
+}
+```
+
+#### `React.Context`
+
+```rescript
+module Context = {
+  let context = React.createContext(() => ())
+
+  module Provider = {
+    let provider = React.Context.provider(context)
+
+    @react.component
+    let make = (~value, ~children) => {
+      React.createElement(provider, {"value": value, "children": children}) // Error
+      // Fix : use the record
+      React.createElement(provider, {value, children}) // OK
+    }
+  }
+}
+```
+
+#### `React.forwardRef`(Discouraged)
+
+```rescript
+module FancyInput = {
+  @react.component
+  let make = React.forwardRef((
+    ~className=?,
+    ~children,
+    ref, // only `ref` is allowed
+  ) =>
+    <div>
+      <input
+        type_="text"
+        ?className
+        ref=?{ref->Js.Nullable.toOption->Belt.Option.map(ReactDOM.Ref.domRef)}
+      />
+      children
+    </div>
+  )
+}
+
+@react.component
+let make = () => {
+  <div>
+    <FancyInput ref=input> // only `ref` is allowed
```

Comment:
@mattdamon108 looks like `input` here is a variable not defined anywhere?
Not sure I understand the example.

Reply:
I added https://github.com/rescript-lang/syntax/pull/652/commits/966b7e7059831fccad1a0553dcae5e94400719e1